handle_multi_pay_invoice_request.go

```diff
@@ -0,0 +1,197 @@
+
+			if !hasPermission {
+				svc.Logger.WithFields(logrus.Fields{
+					"eventId":   event.ID,
+					"eventKind": event.Kind,
+					"appId":     app.ID,
+				}).Errorf("App does not have permission: %s %s", code, message)
+
+				resp, err := svc.createResponse(event, Nip47Response{
+					ResultType: NIP_47_PAY_INVOICE_METHOD,
+					Error: &Nip47Error{
+						Code:    code,
+						Message: message,
+					},
+				}, ss)
+				if err != nil {
+					svc.Logger.WithFields(logrus.Fields{
+						"eventId":        event.ID,
+						"eventKind":      event.Kind,
+						"paymentRequest": invoiceInfo.Invoice,
+						"invoiceId":      invoiceInfo.Id,
+					}).Errorf("Failed to process event: %v", err)
+					return
+				}
+				dTag := []string{"a", fmt.Sprintf("%d:%s:%d", NIP_47_RESPONSE_KIND, event.PubKey, invoiceInfo.Id)}
+				resp.Tags = append(resp.Tags, dTag)
+				svc.PublishEvent(ctx, sub, event, resp)
+				return
+			}
+
+			payment := Payment{App: app, NostrEvent: nostrEvent, PaymentRequest: bolt11, Amount: uint(paymentRequest.MSatoshi / 1000)}
+			insertPaymentResult := svc.db.Create(&payment)
+			if insertPaymentResult.Error != nil {
+				svc.Logger.WithFields(logrus.Fields{
+					"eventId":        event.ID,
+					"eventKind":      event.Kind,
+					"paymentRequest": bolt11,
+					"invoiceId":      invoiceInfo.Id,
+				}).Errorf("Failed to process event: %v", insertPaymentResult.Error)
+				return
+			}
+
+			svc.Logger.WithFields(logrus.Fields{
+				"eventId":   event.ID,
+				"eventKind": event.Kind,
+				"appId":     app.ID,
+				"bolt11":    bolt11,
+			}).Info("Sending payment")
+
+			preimage, err := svc.lnClient.SendPaymentSync(ctx, event.PubKey, bolt11)
+			if err != nil {
+				svc.Logger.WithFields(logrus.Fields{
+					"eventId":   event.ID,
+					"eventKind": event.Kind,
+					"appId":     app.ID,
+					"bolt11":    bolt11,
+				}).Infof("Failed to send payment: %v", err)
+				// TODO: What to do here?
+				nostrEvent.State = NOSTR_EVENT_STATE_HANDLER_ERROR
+				svc.db.Save(&nostrEvent)
+
+				resp, err := svc.createResponse(event, Nip47Response{
+					ResultType: NIP_47_PAY_INVOICE_METHOD,
+					Error: &Nip47Error{
+						Code:    NIP_47_ERROR_INTERNAL,
+						Message: fmt.Sprintf("Something went wrong while paying invoice: %s", err.Error()),
+					},
+				}, ss)
+				if err != nil {
+					svc.Logger.WithFields(logrus.Fields{
+						"eventId":        event.ID,
+						"eventKind":      event.Kind,
+						"paymentRequest": invoiceInfo.Invoice,
+						"invoiceId":      invoiceInfo.Id,
+					}).Errorf("Failed to process event: %v", err)
+					return
+				}
+				dTag := []string{"a", fmt.Sprintf("%d:%s:%d", NIP_47_RESPONSE_KIND, event.PubKey, invoiceInfo.Id)}
+				resp.Tags = append(resp.Tags, dTag)
+				svc.PublishEvent(ctx, sub, event, resp)
+				return
+			}
+			payment.Preimage = &preimage
+			// TODO: What to do here?
+			nostrEvent.State = NOSTR_EVENT_STATE_HANDLER_EXECUTED
```

Comment:
Hmm, we need to review these. They do not apply well to the multi methods.

I would probably set it to executed if at least one payment succeeds, otherwise error. What do you think?

Reply:
CC @bumi

Reply:
> I would probably set it to executed if at least one payment succeeds, otherwise error.

One thought I had was to set a counter to see how many were successful and update this according to that